R/plumber.R

```diff
@@ -887,9 +887,20 @@ Plumber <- R6Class(
     #' @param api This can be
     #'   * an OpenAPI Specification formatted list object
     #'   * a function that accepts the OpenAPI Specification autogenerated by `plumber` and returns a OpenAPI Specification formatted list object.
+    #'   * a path to an OpenAPI Specification
     #'
     #'  The value returned will not be validated for OAS compatibility.
     setApiSpec = function(api = NULL) {
+      if (is.character(api)) {
+        if (tools::file_ext(api) %in% c("yaml", "yml")) {
+          if (!requireNamespace("yaml", quietly = TRUE)) {
+            stop("yaml must be installed to read yaml format")
+          }
+          api <- yaml::read_yaml(api)
```

Comment:
Want to be explicit on this variable.
```suggestion
          api <- yaml::read_yaml(api, eval.expr = FALSE)
```